### DISCUSSION TITLE
Getting started

### DISCUSSION THREAD

**Discussion by w-r-z-k** [scalameta/nvim-metals#141] [Q&A] [answered]
I'm finding it difficult to get started.  The documentation says the following.  But the question I have is: where am I suppose to have these statements?

```
Once installed, the most basic setup is to have the following 
    if has('nvim-0.5')
        augroup lsp
          au!
         au FileType scala,sbt lua require('metals').initialize_or_attach({})
        augroup end
    endif

Or simply as a |vim.cmd| if using a pure Lua setup.

    cmd [[augroup lsp]]
    cmd [[au!]]
    cmd [[au FileType scala,sbt lua require("metals").initialize_or_attach(metals_config)]]
    cmd [[augroup end]]
```

If I put them in init.lua I get errors.

**Comment by ckipp01** [comment, marked as answer]
Hey @w-r-z-k so you're actually mixing the two examples it seems. I'd recommend either going the full vimscript route (what you have up to), or the full Lua example.

The most very minimal setup would simply be:
```lua
local cmd = vim.cmd

cmd [[augroup lsp]]
cmd [[autocmd!]]
cmd [[autocmd FileType scala setlocal omnifunc=v:lua.vim.lsp.omnifunc]]
cmd [[autocmd FileType scala,sbt lua require("metals").initialize_or_attach({})]]
cmd [[augroup end]]
```

Notice the `{}` being passed into `initialize_or_attatch`. More than likely you won't want that to be empty, so much of the other stuff you see in the example in [here](https://github.com/scalameta/nvim-metals/discussions/39) is customizing your config and setting up mappings. Hope that helps.

**Reply by w-r-z-k** [reply]
Thanks.  My problem is that I know Scala and vim but I'm not a neovim/lua configuration expert.  So in looking for an open source IDE, I found this project.  But I find it difficult to bootstrap from there. 

I was able to get it to work by loading neovim from scratch (i.e. `brew install --HEAD neovim`) then copying your minimal init.lua to `.config/nvim/init.lua`.  Then to this file I added the Packer bootstrap after the map function in the init.lua:

```
local execute = vim.api.nvim_command
local fn = vim.fn

local install_path = fn.stdpath('data')..'/site/pack/packer/start/packer.nvim'

if fn.empty(fn.glob(install_path)) > 0 then
  fn.system({'git', 'clone', 'https://github.com/wbthomason/packer.nvim', install_path})
  execute 'packadd packer.nvim'
end
```

I also corrected a typo at the end of the init.lua from:

`Metals_config.on_attach = function(client, bufnr)`

to:

`metals_config.on_attach = function(client, bufnr)`

After that I started nvim and did a `:PackerCompile` and a `:PackerSync`.

And then I did a `:MetalsInstall`

I then figured out that the leader key was set to the default which is backslash. 

At that point, I tried out many of the commands in the init.lua.  Some worked.  Others didn't seem to do anything.

**Reply by ckipp01** [reply]
Hey @w-r-z-k, do you have any example of the commands that didn't work? And to confirm, so did you get Metals started then or not?

**Reply by w-r-z-k** [reply]
Once I got my tools versions sorted out, the whole thing worked.  Metals is started and working.

Now I just have to learn how to use it :)

Thanks very much!
